src/libs/Notification/PushNotification/subscribePushNotification/index.ts

```diff
@@ -50,62 +68,76 @@ function applyOnyxData({reportID, reportActionID, onyxData, lastUpdateID, previo
-    Log.info('[PushNotification] Navigating to report', false, {reportID, reportActionID});
-
-    const policyID = lastVisitedPath && extractPolicyIDFromPath(lastVisitedPath);
-    const report = getReport(reportID.toString());
-    const policyEmployeeAccountIDs = policyID ? getPolicyEmployeeAccountIDs(policyID) : [];
-    const reportBelongsToWorkspace = policyID && !isEmptyObject(report) && doesReportBelongToWorkspace(report, policyEmployeeAccountIDs, policyID);
-
-    Navigation.isNavigationReady()
-        .then(Navigation.waitForProtectedRoutes)
-        .then(() => {
-            // The attachment modal remains open when navigating to the report so we need to close it
-            Modal.close(() => {
-                try {
-                    // If a chat is visible other than the one we are trying to navigate to, then we need to navigate back
-                    if (Navigation.getActiveRoute().slice(1, 2) === ROUTES.REPORT && !Navigation.isActiveRoute(`r/${reportID}`)) {
-                        Navigation.goBack();
-                    }
+    Log.info('[PushNotification] Adding push notification updates to deferred updates queue', false, {reportID, reportActionID});
 
-                    Log.info('[PushNotification] onSelected() - Navigation is ready. Navigating...', false, {reportID, reportActionID});
-                    if (!reportBelongsToWorkspace) {
-                        Navigation.navigateWithSwitchPolicyID({route: ROUTES.HOME});
-                    }
-                    Navigation.navigate(ROUTES.REPORT_WITH_ID.getRoute(String(reportID)));
-                } catch (error) {
-                    let errorMessage = String(error);
-                    if (error instanceof Error) {
-                        errorMessage = error.message;
-                    }
+    const updatesAppliedPromise = Airship.push.getActiveNotifications().then((notifications) => {
+        const onyxUpdates = notifications.map((notification) => extractDataFromPushNotification(notification));
+        // const onyxData = onyxUpdates.reduce<OnyxServerUpdate[]>((onyxUpdatesAcc, onyxUpdate) => (onyxUpdate.onyxData ? [...onyxUpdatesAcc, ...onyxUpdate.onyxData] : onyxUpdatesAcc), []);
 
-                    Log.alert('[PushNotification] onSelected() - failed', {reportID, reportActionID, error: errorMessage});
-                }
+        // const lastUpdateID = onyxUpdates.reduce((max, {lastUpdateID: lastUpdateIDIter}) => Math.max(max, lastUpdateIDIter ?? 0), 0);
+        // const previousUpdateID = onyxUpdates.find((update) => update.lastUpdateID === lastUpdateID)?.previousUpdateID ?? 0;
+
+        // const updates = wrapAirshipUpdatesDataInOnyxUpdatesFromServer({onyxData, lastUpdateID, previousUpdateID});
+        // return DeferredOnyxUpdates.enqueueAndProcess(updates);
+
+        return Promise.all(onyxUpdates.map((update) => applyOnyxData(update)));
+    });
+
+    return updatesAppliedPromise.then(() => {
```

Comment:
Since applying all of the onyx updates can take time (especially when there are gaps detected) we shouldn't block navigation on it. Let's just navigate to the report immediately and do our best to get the data applied ASAP.